File: src/algorithms/find-lucky.js
/**
 * A lucky integer is an integer that has a frequency in the array that is
 * equal to its value.
 * 
 * Given an array of integers arr, returns the largest lucky integer in the
 * array. If there is no lucky integer, returns -1.
 * 
 * @param {array} arr - An array of numbers
 * @returns {number} - Returns the largest lucky integer
 */
function findLucky (arr) {
  let res = -1;
  let frequency = {};
  for (const int of arr) {
    frequency[int] = (frequency[int] || 0) + 1;
  }
  for (const [key, val] of Object.entries(frequency)) {
    if (Number(key) === val && val > res) {
      res = val;
    }
  }
  return res;
}

File: src/algorithms/find-lucky.py
def find_lucky(arr):
    """Given an array of integers arr, a lucky integer is an integer that has a 
    frequency in the array equal to its value. Return the largest lucky integer 
    in the array. If there is no lucky integer return -1.

    Args:
        arr (list[int]): A list of integers.

    Returns:
        int: Returns the largest lucky integer.
    """
    obj = {}
    lucky_int = -1
    for num in arr:
        if num not in obj:
            obj[num] = 1
        else:
            obj[num] += 1
    for key, val in obj.items():
        if key == val and key > lucky_int:
            lucky_int = key
    return lucky_int
